Add release notes for 2.13.0 (#10666)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## Release 2.13.0

> ### Introducing `Global concurrency limits`

Reviewer: ```suggestion
### Introducing global concurrency limits
```

> Control task execution and system stability with Prefect's new Global Concurrency Limits and Rate Limits. 

Reviewer: ```suggestion
Control task execution and system stability with Prefect's new Global Concurrency and Rate Limits. 
```

- **Concurrency Limits:** Manage task execution efficiently, controlling how many tasks can run simultaneously. Ideal for optimizing resource usage and customizing task execution.

- **Rate Limits:** Ensure system stability by governing the frequency of requests or operations. Perfect for preventing overuse, ensuring fairness, and handling errors gracefully.

> Choose Concurrency Limits for resource optimization and task management, and opt for Rate Limits to maintain system stability and fair access to services. To begin using global concurrency limits check out our [guide](https://docs.prefect.io/guides/global-concurrency-limits/).

Reviewer 2: the Prefect concepts should be lowercased

> ### Introducing work pool and worker status

Reviewer 2: we can reference the documentation for this

Workers can now have one of two statuses:
- `Online` - the worker is polling the work pool and is ready to accept work.
- `Offline` - the worker is not polling the work pool and is not ready to accept work. Indicates that the process running the worker has stopped or crashed.

![worker table showing status](https://user-images.githubusercontent.com/12350579/265815336-c8a03c06-2b48-47c5-be93-1dbde0e5bf0d.png)

With the introduction of work pool and worker status, we are deprecating work queue health. Work queue health indicators will be removed in a future release.

See the following pull request for details:
- https://github.com/PrefectHQ/prefect/pull/10636

### Removing deprecated Orion references

Six months ago, we deprecated references to `orion` in our codebase. In this release, we're removing those references. If you still have references to `ORION` in your profile, run `prefect config validate` to automatically convert all of the settings in your *current* profile to the new names!

For example:
```bash
❯ prefect config validate
Updated 'PREFECT_ORION_DATABASE_CONNECTION_URL' to 'PREFECT_API_DATABASE_CONNECTION_URL'.
Configuration valid!
```

> #### Below is a full guide to the changes:

Reviewer 2: @billpalombi Are we okay with the "full guide" here?

Reviewer: I don't know where else we would put it, so I think here is fine.

    - `PREFECT_ORION_TASK_CACHE_KEY_MAX_LENGTH` → `PREFECT_API_TASK_CACHE_KEY_MAX_LENGTH`
    - `PREFECT_ORION_SERVICES_CANCELLATION_CLEANUP_ENABLED` → `PREFECT_API_SERVICES_CANCELLATION_CLEANUP_ENABLED`
##### Changes:
    - Module `prefect.client.orion` → `prefect.client.orchestration`
    - Command group `prefect orion` → `prefect server`
    - Module `prefect.orion` → `prefect.server`
    - Logger `prefect.orion` → `prefect.server`
    - Constant `ORION_API_VERSION` → `SERVER_API_VERSION`
    - Kubernetes deployment template application name changed from `prefect-orion` → `prefect-server`
    - Command `prefect kubernetes manifest orion` → `prefect kubernetes manifest server`
    - Log config handler `orion` → `api`
    - Class `OrionLogWorker` → `APILogWorker`
    - Class `OrionHandler` → `APILogHandler`
    - Directory `orion-ui` → `ui`
    - Class `OrionRouter` → `PrefectRouter`
    - Class `OrionAPIRoute` → `PrefectAPIRoute`
    - Class `OrionDBInterface` → `PrefectDBInterface`
    - Class `OrionClient` → `PrefectClient`

See the following pull request for details:
- Remove deprecated `orion` references — https://github.com/PrefectHQ/prefect/pull/10642

### Fixes

> - Fix an issue with `prefect server start` in Windows - https://github.com/PrefectHQ/prefect/pull/10547

Reviewer 2: ```suggestion
- Fix an issue with `prefect server start` on Windows - https://github.com/PrefectHQ/prefect/pull/10547
```

### Documentation
- Add Kubernetes guide for deploying worker to Azure AKS — https://github.com/PrefectHQ/prefect/pull/10575
- Fix broken link to docker guide with redirect and harmonize naming — https://github.com/PrefectHQ/prefect/pull/10624
- Add information on `--no-prompt` and `PREFECT_CLI_PROMPT` to deployment documentation — https://github.com/PrefectHQ/prefect/pull/10600

> - Remove link to nowhere in api-keys.md — https://github.com/PrefectHQ/prefect/pull/10658

Reviewer 2: ```suggestion
- Remove invalid link in API keys documentation — https://github.com/PrefectHQ/prefect/pull/10658
```

> - Update quickstart screenshots and add cli log output — https://github.com/PrefectHQ/prefect/pull/10659

Reviewer 2: ```suggestion
- Update screenshots and CLI log output in quickstart documentation — https://github.com/PrefectHQ/prefect/pull/10659
```

> - Add documentation for global concurrency limits — https://github.com/PrefectHQ/prefect/pull/10496

Reviewer 2: this can move to highlight

> - Update deployment concept documentation to emphasize server-side deployment — https://github.com/PrefectHQ/prefect/pull/10615

Reviewer 2: I think this could be moved up

> - Add work pool and worker status documentation — https://github.com/PrefectHQ/prefect/pull/10654

Reviewer 2: I think this can be moved to the highlight

> **All changes**: https://github.com/PrefectHQ/prefect/compare/2.12.1...preview

Reviewer 2: ```suggestion
**All changes**: https://github.com/PrefectHQ/prefect/compare/2.12.1...2.13.0
```

## Release 2.13.0

### Introducing global concurrency limits

Control task execution and system stability with Prefect's new global concurrency and rate limits. 

- **Concurrency Limits:** Manage task execution efficiently, controlling how many tasks can run simultaneously. Ideal for optimizing resource usage and customizing task execution.

- **Rate Limits:** Ensure system stability by governing the frequency of requests or operations. Perfect for preventing overuse, ensuring fairness, and handling errors gracefully.

Choose concurrency limits for resource optimization and task management, and opt for rate limits to maintain system stability and fair access to services. To begin using global concurrency limits check out our [guide](https://docs.prefect.io/guides/global-concurrency-limits/).

See the following pull request for details:
- https://github.com/PrefectHQ/prefect/pull/10496

### Introducing work pool and worker status

Work pools and workers are critical components of Prefect's distributed execution model. To help you monitor and manage your work pools and workers, we've added status indicators to the Prefect UI.

Work pools can now have one of three statuses:

> - `Ready` -  at least on online worker is polling the work pool and the work pool is ready to accept work.

Reviewer 2: ```suggestion
- `Ready` -  at least one online worker is polling the work pool and the work pool is ready to accept work.
```
